File: core/src/main/java/heroes/journey/entities/actions/inputs/ActionInput.java
package heroes.journey.entities.actions.inputs;

import heroes.journey.GameState;
import lombok.Getter;

import java.util.UUID;

@Getter
public class ActionInput {

    private final GameState gameState;
    private final UUID entityId;

    public ActionInput(GameState gameState, UUID entityId) {
        this.gameState = gameState;
        this.entityId = entityId;
    }

}

File: core/src/main/java/heroes/journey/entities/actions/ActionInput.kt
package heroes.journey.entities.actions

import heroes.journey.GameState
import java.util.*

class ActionInput(val gameState: GameState, val entityId: UUID? = null, input: Map<String, String>) :
    HashMap<String, String>() {

    constructor(gameState: GameState) : this(gameState, null, hashMapOf())
    constructor(gameState: GameState, entityId: UUID) : this(gameState, entityId, hashMapOf())

    init {
        for ((key, value) in input) {
            this[key] = value
        }
    }

    fun getInput(): String? {
        return this["input"]
    }

    fun hasInput(): Boolean {
        return getInput() != null && getInput() != ""
    }

    fun getHashMapCopy(): HashMap<String, String> {
        val copy: HashMap<String, String> = HashMap()
        for ((key, value) in this) {
            copy[key] = value
        }
        return copy
    }
}
